### PR TITLE
Replace chrono with time due to RUSTSEC-2020-0159

### LINKED ISSUE #308: Use time crate instead of chrono
#### Use Case:

The chrono crate is unmaintained and has a [known security flaw](https://rustsec.org/advisories/RUSTSEC-2020-0159) (albeit one not triggered by nats.rs), requiring whitelisting in security tools like `cargo audit`. Changing to use the time crate would avoid this needed whitelisting.

#### Proposed Change:

Use the time crate in place of chrono. This does require a breaking API change.

#### Who Benefits From The Change(s)?

Users who use tools such as `cargo audit` to check their dependency chains for known security issues.

#### Alternative Approaches

Stay as is and document prominently that the security flaw does not affect nats.rs.

#### Notes

I have a patch available for this. Would copyright assignment or similar need to happen?

### PR DESCRIPTION
**Pull request by ShellWowza**
Resolves #308.

### CODE REVIEW

File: src/jetstream/types.rs
Comment (caspervonb) [inline src/jetstream/types.rs:17]: Is there no crate for this ser/de in the wild?

Comment (ShellWowza) [reply src/jetstream/types.rs:17]: I have just discovered that time has a `serde-well-known` feature for this. I'll rework this to use that.

**Review by caspervonb — APPROVED** [review]
LGTM!

I'll let @Jarema have a look before I squash and merge.

**Review by Jarema — APPROVED** [review]
LGTM!

Thanks for the contribution!

**Pull request merged**
